sdk/python/feast/expediagroup/vectordb/milvus_online_store.py

```diff
@@ -296,3 +315,67 @@ def _format_data_for_milvus(self, feast_data):
 
         transformed_data = [list(item) for item in zip(*milvus_data)]
         return transformed_data
+
+    def _create_index_params(
+        self, index_type: MilvusIndexType, tags: Dict[str, str], data_type: DataType
+    ):
+        """
+        Parses the tags to generate the index_params needed to create the specified index
+
+        Parameters:
+            index_type (MilvusIndexType): the index type to be created
+            tags (Dict): the tags associated with the field
+            data_type (DateType): the data type of the field
+
+        Returns:
+            (Dict): a dictionary formatted for the create_index params argument
+        """
+
+        params = {}
+        metric_type = "L2"
+
+        def get_required_from_tags(*keys):
+            for k in keys:
+                v = tags.get(k)
+                if v:
+                    return int(v)
+            raise KeyError(
+                f"tags are required to have one of {keys} for index_type {index_type.name}"
+            )
+
+        if index_type in [
+            MilvusIndexType.IVFLAT,
+            MilvusIndexType.IVF_SQ8,
+            MilvusIndexType.IVF_SQ8H,
+            MilvusIndexType.IVF_PQ,
+        ]:
+            params["nlist"] = get_required_from_tags("nlist", "n_list")
+
+        if index_type is MilvusIndexType.IVF_PQ:
```

Comment:
I think we should rather just let a user pass "index_params" that we then pass to the Milvus SDK when creating the index. If we check it here there is little benefit because if a wrong parameter is passed it would fail soon after this anyway. Bhargav and I discussed and agreed on moving validation to the pydantic model so that invalid inputs would fail at the time of registration. However, checking which index type requires which parameters is overkill unless there is an easy way to get this through the Milvus SDK and hence wouldn't require much maintenance in case parameters change, index types are added, etc.